### PR TITLE
Pg gate entities

### PR DESCRIPTION
**Pull request by johnfangAFW**
Add entities classes for PG gate and Catalog manager. Some classes might be refactored further later. This PR will unblock the catalog manager work on table operations.

### CODE REVIEW

File: src/gausskernel/storage/access/k2/k2pg_util.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 using k2pg::K2PgErrorCode;
 
-extern "C" {
+//extern "C" {
```
Comment (iafuture) [inline src/gausskernel/storage/access/k2/k2pg_util.cpp]: any reason to keep this?

File: src/gausskernel/storage/access/k2/k2pg_util.cpp [hunk 2]
```diff
@@ -42,15 +42,15 @@ K2PgStatus K2PgInit(const char* argv0,
...
       .msg = "OK",
       .detail = "K2PgInit OK"
   };
-  
+
   return status;
 }
 
-} // extern "C"
+//} // extern "C"
```
Comment (iafuture) [inline src/gausskernel/storage/access/k2/k2pg_util.cpp]: ... and this

File: src/gausskernel/storage/access/k2/catalog/k2_catalog_defaults.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*
+MIT License
+
+Copyright(c) 2020 Futurewei Cloud
```
Comment (iafuture) [inline src/gausskernel/storage/access/k2/catalog/k2_catalog_defaults.cpp]: copyright 2022

File: src/include/access/k2/catalog/k2_catalog_defaults.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    static const uint32_t oid_tablecolumn_meta;
+    static const uint32_t oid_indexcolumn_meta;
+
+    static const std::string TABLE_ID_COLUMN_NAME;
+    static const std::string INDEX_ID_COLUMN_NAME;
+    static const std::string BASE_TABLE_ID_COLUMN_NAME;
+
+    static const std::string shared_table_skv_colllection_id;
+
+//    static inline const k2::Duration catalog_manager_background_task_initial_wait = 1s;
```
Comment (iafuture) [inline src/include/access/k2/catalog/k2_catalog_defaults.h]: not needed?

File: src/include/access/k2/catalog/k2_catalog_defaults.h [hunk 5]
```diff
@@ -0,0 +1,65 @@
+/*
+MIT License
+
+Copyright(c) 2020 Futurewei Cloud
```
Comment (iafuture) [inline src/include/access/k2/catalog/k2_catalog_defaults.h]: copyright fix

File: src/include/access/k2/pg_ddl.h [hunk 6]
```diff
@@ -0,0 +1,28 @@
+/*
```
Comment (iafuture) [inline src/include/access/k2/pg_ddl.h:1]: do we need this file?

File: src/include/access/k2/pg_session.h [hunk 7]
```diff
@@ -0,0 +1,60 @@
+/*
```
Comment (iafuture) [inline src/include/access/k2/pg_session.h:1]: I think this should be covered by the k2 txn manager, no?

**Review by iafuture — CHANGES_REQUESTED** [review]

File: src/gausskernel/storage/access/k2/k2pg_util.cpp (hunk 1 above)
Comment (johnfangAFW) [reply src/gausskernel/storage/access/k2/k2pg_util.cpp]: good catch, will remove.

File: src/gausskernel/storage/access/k2/catalog/k2_catalog_defaults.cpp (hunk 3 above)
Comment (johnfangAFW) [reply src/gausskernel/storage/access/k2/catalog/k2_catalog_defaults.cpp]: will do

File: src/include/access/k2/catalog/k2_catalog_defaults.h (hunk 4 above)
Comment (johnfangAFW) [reply src/include/access/k2/catalog/k2_catalog_defaults.h]: Not sure how to use the header files from k2 platform in opengauss, disable them temporarily

File: src/include/access/k2/catalog/k2_catalog_defaults.h (hunk 5 above)
Comment (johnfangAFW) [reply src/include/access/k2/catalog/k2_catalog_defaults.h]: will do

File: src/include/access/k2/pg_ddl.h (hunk 6 above)
Comment (johnfangAFW) [reply src/include/access/k2/pg_ddl.h:1]: will put all ddl related logic into this file and pass the calls to catalog manager. Should be implemented by Ahsan.

File: src/include/access/k2/pg_session.h (hunk 7 above)
Comment (johnfangAFW) [reply src/include/access/k2/pg_session.h:1]: This is the session object related to each database client connection, for example, we need to keep track of the current database for this session. This object could be integrated with k2 txn manager later.

File: src/include/access/k2/catalog/k2_catalog_defaults.h (hunk 4 above)
Comment (johnfangAFW) [reply src/include/access/k2/catalog/k2_catalog_defaults.h]: fixed

**Pull request merged**